keyboards/mschwingen/modelm/led_ws2812/rules.mk

```diff
@@ -0,0 +1,3 @@
+# variant for WS2812 LEDs
+SRC += ws2812.c
```

Comment:
Why this, instead of rgblight? 

There is a "light layers" feature that is designed for adding indicators. 
https://docs.qmk.fm/#/feature_rgblight?id=lighting-layers

Reply:
rgblight seems to add a lot of ballast when all I need is three indicator LEDs without any backlight features. I will look into using lightling layers, but at first look, I am not quite sure if it can do what I need. Also, I still need similar code for the non-WS2812 version.

Reply:
I tried, but I can't get this to do what I want. 
I can get this partially working using lightling-layers, but some things don't work (manually changing a layer using shift-shift-number does not activate the layer_state_set_user callback), and the firmware is already 2000 bytes larger than using my original solution, nearing a full flash. 

Also, the functionality I want (state indicator LEDs that indicate caps/num/scroll lock state, but change their color based on layer) does not really work well using the stacked layer approach.

To get the full functionality of my current code, I would have to use one layer per indicator LED, and then enable/disable that layer to turn on/off the LED - this is wasteful, I would rather directly control the LED as in the current patch.